### PR TITLE
Homework2

### PR DESCRIPTION
**Pull request by krzkuc1985**

### CODE REVIEW

File: module2/homework/generateSequence/generateSequence.hpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+std::vector<int> generateSequence(int count, int step) {
+    if (count <= 0) {
+        return {};
+    }
+    std::vector<int> vec;    
+    for (size_t i = 1; i <= count; i++) {
+        vec.push_back(i*step);
+    }
+    return vec;
+}
```
Comment (danielzwierz) [inline module2/homework/generateSequence/generateSequence.hpp:13]: Add new line at end of file

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include <vector>
+#include <limits>
+
+int maxOfVector(const std::vector<int>& vec) {
+    int max = vec.front();
+    for (const auto& element : vec) {
+        if (max < element) {
+        max = element;
```
Comment (danielzwierz) [inline module2/homework/maxOfVector/maxOfVector.hpp:9]: wrong indentation

File: module2/homework/maxOfVector/maxOfVector.hpp [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        }
+    }
+    return max;
+}
```
Comment (danielzwierz) [inline module2/homework/maxOfVector/maxOfVector.hpp:13]: Add new line at end of file

File: module2/homework/nwdNww/nwdNww.hpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    }
+    return abs(lhs);
+}
+
+int NWW(int lhs, int rhs) {
+    if (lhs == 0 || rhs == 0) {
+        return 0;
+    }
+    lhs = abs(lhs);
+    rhs = abs(rhs);
```
Comment (danielzwierz) [inline module2/homework/nwdNww/nwdNww.hpp:19]: you can include abs() in return, 2 line less

File: module2/homework/addEven/addEven.hpp [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+int addEven(const std::vector<int>& numbers) {
+    int sum = 0;
+    for (const auto& element : numbers) {
+        if (!(element % 2)) {
+            sum += element;
+        }
+    }
+    return sum;
+}
```
Comment (danielzwierz) [inline module2/homework/addEven/addEven.hpp:12]: Add new line at end of file

**Comment by krzkuc1985** [comment]
To reduce the number of files, do I need to make a new pull request?

**Comment by ziobron** [comment]
> To reduce the number of files, do I need to make a new pull request?

No. I have changed the base branch into which you wanted to make this Pull Request. It was raised against master, but it should be against module2. You can also do the same when you click Edit next to the name of this PR.

+4 XP addEven
+6 XP NwwNwd
+5 XP maxOfVector
+5 XP generateSequence
+8 XP delivery before 31.05

{ "requirement_ids": [1119, 1631, 1119, 1633, 1119, 1634, 1119, 1635], "points": [2, 4, 2, 6, 2, 5, 2, 5] }

**Pull request closed without merging**